src/Accelerators/NNPA/Conversion/ONNXToZHigh/DevicePlacement.cpp

```diff
@@ -13,13 +13,26 @@
 // - user configuration file if given
 // - a cost model
 //
+// Device placement is done via setting `device` attribute for each operation.
+// Values for `device` attribute is one of the following strings:
+// - "": an empty string means the compiler decides whether the operation is on
+// CPU or NNPA.
+// - "nnpa": the operation may run on NNPA or CPU, and the final decision is
+// made by the compiler. If `device=nnpa` is the result of this device-placement
+// pass, then it means the compiler thinks it is suitable for NNPA.
+// - "cpu": the operation is guaranteed to run on CPU.
```

Comment:
This might not be appropriate question for this PR, but what is the difference between `device=""` and `device=nnpa`? I think compiler determines the device when either attribute is used. By specifying `device=nnpa`, can we get different results, or is this useful for users? I mean only providing `decice=""` and `device=cpu` attribute is not enough?

Reply:
Had the same question, why do we even accept that?

Reply:
- When users **setting** device placement via a JSON file, there is no difference between `device=""` and `device=nnpa`, because we are able to force to CPU at this moment only, meaning `device=cpu`. I hope users can set NNPA in the future  by `device=nnpa` and `device=""` means let the compiler decide, but I haven't had a good solution yet.

- When the compiler **prints out** its device placement, e.g. via `--EmitONNXIR` or `--save-device-placement-file`, `device=""` and `device="nnpa"` are completely different. `device="nnpa"` means the compiler decide to run the op on NNPA while `device=""` means there is no guarantee that the op runs on CPU or NNPA.